packages/synchro-charts/src/components/sc-dial/sc-dial.spec.tsx

```diff
@@ -0,0 +1,243 @@
+    html: '<div></div>',
+    supportsShadowDom: false,
+  });
+  const dial = page.doc.createElement('sc-dial') as CustomHTMLElement<Components.ScDial>;
+  const props: Partial<Components.ScDialBase> = {
+    widgetId: 'test-dial-widget',
+    dataStream: DATA_STREAMS[0],
+    viewport: VIEWPORT,
+    ...propOverrides,
+  };
+  update(dial, props);
+  page.body.appendChild(dial);
+
+  await page.waitForChanges();
+
+  return { page, dial };
+};
+
+describe('Only input data stream', () => {
+  it('Render dial base component', async () => {
+    const laterDate = new Date((new Date(2020, 1, 0, 0) as Date).getTime() + MINUTE_IN_MS);
+    const SOME_LATER_POINT: DataPoint<number> = { y: 111, x: laterDate.getTime() };
+    const { dial } = await newValueSpecPage({ dataStream: { ...DATA_STREAM, data: [SOME_LATER_POINT] } });
+
+    const dialBases = dial.querySelectorAll('sc-dial-base');
+    expect(dialBases.length).toBe(1);
+  });
+
+  it('Only input dataStream', async () => {
+    const laterDate = new Date((new Date(2020, 1, 0, 0) as Date).getTime() + MINUTE_IN_MS);
+    const SOME_LATER_POINT: DataPoint<number> = { y: 111, x: laterDate.getTime() };
+    const DATA = { ...DATA_STREAM, data: [SOME_LATER_POINT] };
+    const { dial } = await newValueSpecPage({ dataStream: DATA });
+
+    const dialBases = dial.querySelectorAll('sc-dial-base');
+    expect(dialBases.length).toBe(1);
+
+    expect(dialBases[0].propertyStream).toBe(DATA);
+    expect(dialBases[0].propertyPoint).toBe(SOME_LATER_POINT);
+    expect(dialBases[0].alarmStream).toBeUndefined();
+    expect(dialBases[0].breachedThreshold).toBeUndefined();
+    expect(dialBases[0].viewport).toBe(VIEWPORT);
+    expect(dialBases[0].valueColor).toBeUndefined();
+    expect(dialBases[0].isLoading).toBe(false);
+  });
+
+  it('DataStream is loading', async () => {
+    const laterDate = new Date((new Date(2020, 1, 0, 0) as Date).getTime() + MINUTE_IN_MS);
+    const SOME_LATER_POINT: DataPoint<number> = { y: 111, x: laterDate.getTime() };
+    const DATA = { ...DATA_STREAM, data: [SOME_LATER_POINT], isLoading: true };
+    const { dial } = await newValueSpecPage({ dataStream: DATA });
+
+    const dialBases = dial.querySelectorAll('sc-dial-base');
+    expect(dialBases.length).toBe(1);
+
+    expect(dialBases[0].propertyStream).toBe(DATA);
+    expect(dialBases[0].propertyPoint).toBe(SOME_LATER_POINT);
+    expect(dialBases[0].alarmStream).toBeUndefined();
+    expect(dialBases[0].breachedThreshold).toBeUndefined();
+    expect(dialBases[0].viewport).toBe(VIEWPORT);
+    expect(dialBases[0].valueColor).toBeUndefined();
+    expect(dialBases[0].isLoading).toBe(true);
+  });
+});
+
+describe('alarm', () => {
+  const ANNOTIONS = {
+    y: [
+      {
+        color: '#000',
+        value: 'Normal',
+        comparisonOperator: COMPARISON_OPERATOR.EQUAL,
+        dataStreamIds: ['some-id'],
+        icon: StatusIcon.NORMAL,
+      },
+    ],
+  };
+  const ASSOCIATEDSTREAMS = [
```

Comment:
Add underscore between words because variables `NAMEDLIKETHIS` are hard to read

Reply:
Done